WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationService.java

```diff
@@ -0,0 +1,338 @@
+                siteSlug,
+                siteTitle,
+                language,
+                SiteStore.SiteVisibility.PUBLIC,
+                false);
+        mDispatcher.dispatch(SiteActionBuilder.newCreateNewSiteAction(newSitePayload));
+        AppLog.i(T.NUX, "User tries to create a new site, title: " + siteTitle + ", SiteName: " + siteSlug);
+
+        return START_REDELIVER_INTENT;
+    }
+
+    private void activateTheme(final SiteModel site, final String themeId) {
+        WordPress.getRestClientUtils().setTheme(site.getSiteId(), themeId, new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                ThemeTable.setCurrentTheme(WordPress.wpDB.getDatabase(), String.valueOf(site.getSiteId()), themeId);
+
+                setState(SiteCreationPhase.SUCCESS);
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                setState(SiteCreationPhase.FAILURE);
+//                ToastUtils.showToast(ThemeBrowserActivity.this, R.string.theme_activation_error,
+//                        ToastUtils.Duration.SHORT);
+            }
+        });
+    }
+
+    // OnChanged events
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onNewSiteCreated(SiteStore.OnNewSiteCreated event) {
+        AppLog.i(T.NUX, event.toString());
+        if (event.isError()) {
+            setState(SiteCreationPhase.FAILURE);
+//            showError(event.error.type, event.error.message);
+            return;
+        }
+
+        AnalyticsTracker.track(AnalyticsTracker.Stat.CREATED_SITE);
+
+        setState(SiteCreationPhase.FETCHING_NEW_SITE);
+
+        mNewSiteRemoteId = event.newSiteRemoteId;
+
+        // We can't get all the site informations from the new site endpoint, so we have to fetch the site list.
+        mDispatcher.dispatch(SiteActionBuilder.newFetchSitesAction());
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSiteChanged(SiteStore.OnSiteChanged event) {
+        AppLog.i(T.NUX, event.toString());
+        if (event.isError()) {
+            // Site has been created but there was a error while fetching the sites. Can happen if we get
+            // a response including a broken Jetpack site. We can continue and check if the newly created
+            // site has been fetched.
+            AppLog.e(T.NUX, event.error.type.toString());
+        }
+
+        final SiteModel site = mSiteStore.getSiteBySiteId(mNewSiteRemoteId);
+
+        if (mSiteCreationPhase == SiteCreationPhase.FETCHING_NEW_SITE) {
+            Intent intent = new Intent();
+            if (site == null) {
+                setState(SiteCreationPhase.FAILURE);
+                //            ToastUtils.showToast(getActivity(), R.string.error_fetch_site_after_creation, ToastUtils.Duration.LONG);
```

Comment:
same here, right? (same question as in [comment above](https://github.com/wordpress-mobile/WordPress-Android/pull/6825/files#r150237854))

Reply:
Yes :)